instrumentation/lettuce/lettuce-5.1/src/main/java/io/opentelemetry/javaagent/instrumentation/lettuce/v5_1/LettuceArgSplitter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.lettuce.v5_1;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public final class LettuceArgSplitter {
+  private static final Pattern KEY_PATTERN =
+      Pattern.compile("((key|value)<(?<wrapped>[^>]+)>|(?<plain>[0-9A-Za-z=]+))(\\s+|$)");
+
+  // this method removes the key|value<...> wrappers around redis keys or values and splits the args
+  // string
+  public static List<String> splitArgs(@Nullable String args) {
+    if (args == null || args.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    List<String> argsList = new ArrayList<>();
+    Matcher m = KEY_PATTERN.matcher(args);
+    while (m.find()) {
+      String wrapped = m.group("wrapped");
+      if (wrapped != null) {
+        argsList.add(wrapped);
+      } else {
+        argsList.add(m.group("plain"));
+      }
+    }
```

Comment:
sorry, i didn't explain my thought about this well, this is more what i was thinking, keeping the previous `KEY_PATTERN`:

please pick whichever implementation you want (original, current, or below), i don't see a clear winner myself

```suggestion
    int lastEnd = 0;
    while (m.find()) {
      int wrapperStart = m.start();
      argsList.addAll(split(args, lastEnd, wrapperStart));
      argsList.add(m.group("key"));
      lastEnd = m.end();
    }
    argsList.addAll(split(args, lastEnd, args.length()));

  ...

  private static final Pattern SPLIT_PATTERN = Pattern.compile(" ");

  private static List<String> split(String args, int from, int to) {
    return Arrays.asList(SPLIT_PATTERN.split(args.subSequence(from, to)));
  }

```

Reply:
Oh, I see. Hmm, I think I'll leave it as it is now - the pattern is a bit more complex but I like the fact that we only go through the entire string once.